libs/community/langchain_community/vectorstores/yellowbrick.py

```diff
@@ -1,204 +1,523 @@
 from __future__ import annotations
 
+import atexit
+import csv
+import enum
 import json
 import logging
+import random
 import uuid
-import warnings
-from itertools import repeat
+from contextlib import contextmanager
+from io import StringIO
 from typing import (
+    TYPE_CHECKING,
     Any,
+    Dict,
+    Generator,
     Iterable,
     List,
     Optional,
     Tuple,
     Type,
+    Union,
 )
 
 from langchain_core.embeddings import Embeddings
 from langchain_core.vectorstores import VectorStore
 
 from langchain_community.docstore.document import Document
 
-logger = logging.getLogger(__name__)
+if TYPE_CHECKING:
+    from psycopg2.extensions import connection as PgConnection
+    from psycopg2.extensions import cursor as PgCursor
 
 
 class Yellowbrick(VectorStore):
     """Yellowbrick as a vector database.
-
     Example:
         .. code-block:: python
             from langchain_community.vectorstores import Yellowbrick
             from langchain_community.embeddings.openai import OpenAIEmbeddings
             ...
     """
 
+    class IndexType(str, enum.Enum):
+        """Enumerator for the supported Index types within Yellowbrick."""
+
+        NONE = "none"
+        LSH = "lsh"
+
+    class IndexParams:
+        """Parameters for configuring a Yellowbrick index."""
+
+        def __init__(
+            self,
+            index_type: Optional["Yellowbrick.IndexType"] = None,
+            params: Optional[Dict[str, Any]] = None,
+        ):
+            if index_type is None:
+                index_type = Yellowbrick.IndexType.NONE
+            self.index_type = index_type
+            self.params = params or {}
+
+        def get_param(self, key: str, default: Any = None) -> Any:
+            return self.params.get(key, default)
+
     def __init__(
         self,
         embedding: Embeddings,
-        connection_string: str,
```

Comment:
this is a breaking change — are we sure we want to make it?

Reply:
I've decided to remove the breaking change here and leave adding the ability to pass in an existing connection to a later PR.